ValidationMixin.js

```diff
@@ -1,83 +1,53 @@
-"use strict";
+'use strict';
 
-var assign = Object.assign || require('object.assign');
+require('object.assign').shim();
+var result = require('lodash.result');
 var ValidationFactory = require('./ValidationFactory');
 
 var ValidationMixin = {
-
   /**
-   * Validate single form field against the components state. If no field
-   * is provided, validate entire form.
-   *
-   * @param {?String} field State key to validate
-   * @return {Object} newly updated errors object keyed on state field
-   * names. Missing key or undefined value indicates no error.
+   * Check for sane configurations
    */
-  validate: function(field) {
-    var validatorTypes = this.validatorTypes || {};
-    if (typeof this.validatorTypes === 'function') {
-      validatorTypes = this.validatorTypes();
+  componentDidMount: function() {
+    if (typeof this.validatorTypes !== 'function' && !Array.isArray(this.validatorTypes)) {
+      throw Error('invalid `validatorTypes` type');
     }
-    var options = {
-      schema: validatorTypes,
-      state: this.state,
-      field: field
-    };
-    var nextErrors = assign({}, this.state.errors, ValidationFactory.validate(options));
-    this.setState({
-      errors: nextErrors
-    });
-    return nextErrors;
   },
 
   /**
-   * Convenience method to validate a field via an event handler. Useful for
-   * onBlur, onClick, onChange, etc...
+   * Validate single form field or entire form against the component's state.
    *
-   * @param {?String} field State key to validate
-   * @param {?Boolean} preventDefault flag to indicate that this event should be canceled.
-   * default is false.
-   * @return {function} validation event handler
+   * @param {?String} key. Field key to validate (entire form if undefined).
+   * @return {Boolean} Result of `isValid` call after validation.
    */
-  handleValidation: function(field, preventDefault) {
-    return function(event) {
-      if (preventDefault === true) {
-        event.preventDefault();
-      }
-      this.validate(field);
-    }.bind(this);
+  validate: function(key) {
+    var schema = result(this, 'validatorTypes', {});
+    var errors = Object.assign({}, this.state.errors, ValidationFactory.validate(this.state, schema, key));
+    this.setState({
+      errors: errors
+    });
+    return this.isValid(key);
```

Comment:
When we call setState() from `validate()` this is an ansync call, which means the call to `isValid()` is not guaranteed to have the new state to validate against.